### PR TITLE
Make the verify ballot encryption only run for admin and not for guar…

### LINKED ISSUE #757: ✨ Perf enhancement for Tally
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

The guardians and the admin all do a verify_ballot_encryption to go over all of the encryptions in each of the ballots. There is no need for this to be done 3+ times when a single check is enough.  

The guardians do not need to verify the encryptions and this will save time.

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by SteveMaier-IRT**
…dians

### Issue
Fixes #757 

### Description
Made the verify ballot encryption calls only run from the admin during the tally.

### Testing
Tests have been updated.

### CODE REVIEW

**Review by lprichar — COMMENTED** [review]
❤️ the perf speedup

File: src/electionguard/tally.py [hunk 1]
```diff
@@ -256,6 +256,7 @@ def append(
     def batch_append(
         self,
         ballots: Iterable[Tuple[Any, SubmittedBallot]],
+        isAdmin: bool,
```
[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply src/electionguard/tally.py:259]: updated.

File: src/electionguard_gui/services/decryption_stages/decryption_stage_base.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ def get_tally(
     manifest: Manifest,
     context: CiphertextElectionContext,
     ballots: List[SubmittedBallot],
+    isAdmin: bool,
```
[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply src/electionguard_gui/services/decryption_stages/decryption_stage_base.py:62]: done

File: src/electionguard/tally.py [hunk 3]
```diff
@@ -268,7 +269,7 @@ def batch_append(
             # get the value of the dict
             ballot_value = ballot[1]
             if not self.__contains__(ballot) and ballot_is_valid_for_election(
-                ballot_value, self._internal_manifest, self._encryption
+                ballot_value, self._internal_manifest, self._encryption, isAdmin
```
[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply src/electionguard/tally.py:272]: The problem is that the ballot_is_valid_for_election(...) does more than just check for the validity of the encryption so it needs to be called no matter what to get the rest of the functionality.  We want to have it check that the ballot style is correct and anything else in there.

[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply src/electionguard/tally.py:272]: This simplification will be added to a future change where the call to check the encryption is moved.

File: tests/property/test_tally.py [hunk 4]
```diff
@@ -315,7 +315,7 @@ def _cannot_erroneously_mutate_state(
         first_ballot_hash = ballot.manifest_hash
         ballot.manifest_hash = ONE_MOD_Q
         self.assertIsNone(tally_ballot(ballot, tally))
-        self.assertFalse(tally.append(ballot))
+        self.assertFalse(tally.append(ballot, True))
```
[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply tests/property/test_tally.py:318]: done

File: src/electionguard/tally.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lprichar) [reply src/electionguard/tally.py]: Good point.  Looking forward to refactoring that into two methods at some point.  Disregard this comment.

**Review by lprichar — APPROVED** [review]

**Pull request merged**